### PR TITLE
[feat] Support global index for obkv

### PR DESCRIPTION
**Pull request by GroundWu**
## Summary
- support global index and relate test cases for obkv

## Solution Description
- get and calculate global index route info
- add dml and query test cases

### CODE REVIEW

File: protocol/index_type.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	IndexTypeIsNot ObIndexType = iota
+	IndexTypeNormalLocal
+	IndexTypeUniqueLocal
+	IndexTypeNormalGlobal
+	IndexTypeUniqueGlobal
+	IndexTypePrimary
+	IndexTypeDomainCtxcat
+	IndexTypeNormalGlobalLocalStorage
+	IndexTypeUniqueGlobalLocalStorage
+	IndexTypeSpatialLocal              = 10
```
Comment (IHEII) [inline protocol/index_type.go:32]: It seems that we do not need to assign  since we use `iota`

File: route/location.go [hunk 2]
```diff
@@ -748,3 +749,115 @@ func fetchSubPart(ctx context.Context, db *DB, partFuncType obPartFuncType, entr
...
+	)
+	if err != nil {
+		return 0, errors.WithMessagef(err, "new db, sysUA:%s, addr:%s", sysUA.String(), addr.String())
+	}
+	defer func() {
+		_ = db.Close()
+	}()
+
+	// 2. make sql
+	sql := fmt.Sprintf("select table_id from oceanbase.__all_virtual_proxy_schema "+
```
Comment (IHEII) [inline route/location.go:777]: Make this SQL a const string listed above this file.

File: route/location.go [hunk 3]
```diff
@@ -748,3 +749,115 @@ func fetchSubPart(ctx context.Context, db *DB, partFuncType obPartFuncType, entr
...
+	)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "new db, sysUA:%s, addr:%s", sysUA.String(), addr.String())
+	}
+	defer func() {
+		_ = db.Close()
+	}()
+
+	// 2. make sql
+	sql := fmt.Sprintf("select data_table_id, table_id, index_type from oceanbase.__all_virtual_table where table_name = '%s'", indexTableName)
```
Comment (IHEII) [inline route/location.go:828]: Make this SQL a const string listed above this file.

File: route/route_info.go [hunk 4]
```diff
@@ -46,6 +46,9 @@ type ObRouteInfo struct {
 	tableRoster      ObTableRoster
 	serverRoster     ObServerRoster // all servers which contain current tenant
 	taskInfo         *ObRouteTaskInfo
+
+	indexMutexes sync.Map // map[indexName]sync.RWMutex
+	indexRoster  sync.Map // map[indexName]
```
Comment (IHEII) [inline route/route_info.go:51]: I guess we have not explained well here.

**Comment by WeiXinChan** [comment]
LGTM

**Pull request merged**